libqfieldsync/project_checker.py

```diff
@@ -439,3 +444,44 @@ def check_external_layers(
             )
 
         return None
+
+    def check_supported_datasource_for_cloud(
+        self, layer_source: LayerSource
+    ) -> Optional[FeedbackResult]:
+        """Check if offline editing layer is not a supported format for cloud"""
+        layer = layer_source.layer
+
+        # Check if layer is marked for offline editing
+        if layer_source.cloud_action != SyncAction.OFFLINE:
+            return None
+
+        # Check if layer is read-only
+        if layer.readOnly():
+            return None
+
+        # Check if layer is in a supported format
+        provider_type = layer.providerType()
+        filename = layer_source.filename
+
+        # Check if it's a GeoPackage
+        if layer.dataProvider().storageType() == "GPKG":
+            return None
+
+        # Check if it's a PostgreSQL database
+        if provider_type == "postgres":
+            return None
+
+        # Check if it's a non-spatial delimited text file
+        if provider_type == "delimitedtext" and (
+            filename.endswith(".csv") or filename.endswith(".txt")
```

Comment:
You should use some other way to check whether the file is a CSV file. Note that there are two ways to open CSV in QGIS.
- the delimited text
- GDAL (drag n drop a CSV in QGIS)